### ISSUE TITLE
hasStyle: incorrect child components check in case of nested CSS

### ISSUE THREAD

**Issue by matrunchyk** [eddyerburgh/avoriaz#25]
**UPD. The real description of the problem is [here](https://github.com/eddyerburgh/avoriaz/issues/25#issuecomment-293096890)**.

Hi guys, I've just noticed that `zIndex` doesn't work with `hasStyle`.

![Alt text](https://monosnap.com/file/VgLx5mfyH0u2KHIAKs8wsUoGDCJO0c.png)

I tried also `.hasStyle('z-index', '9999')`. Still `false`.

What am I doing wrong? Thanks.

**Comment by eddyerburgh** [comment]
Hi @matrunchyk ,

hasStyle uses the same values as css, so you use `z-index` instead of the JavaScript version `zIndex`.

This should work: 

```javascript
expect(temp1.hasStyle('z-index', '9999').to.equal(true)
```

**Comment by matrunchyk** [comment]
As I mentioned, `.hasStyle('z-index', '9999')` doesn't work either.
And even more. It works very weird :)

![Alt text](https://monosnap.com/file/Tz10sx3Q4itJDXO3k2bCOojjxXoiW5.png)

**Comment by matrunchyk** [comment]
Ok, so I figured out the problem.
My component's CSS has rules like that:

```
.MyComponent > div {
    color: red;
}

```
As you can see, that div would be red only within .MyComponent.

But [here](https://github.com/eddyerburgh/avoriaz/blob/master/src/Wrapper.js#L174) you move an element (div in this case) to the body. So now it's impossible to check a read style of the div.

I'd suggest to not moving that only particular div (or any other child), but mount a whole component at all.

**Comment by eddyerburgh** [comment]
Thanks, I'll look into this tonight.

**Comment by eddyerburgh** [comment]
Can you link me to the component and test?

**Comment by matrunchyk** [comment]
@eddyerburgh Created a small repo with a component and a test https://github.com/matrunchyk/avoriaz-issue-25

**Comment by matrunchyk** [comment]
Alright, thank you!

**Comment by eddyerburgh** [comment]
Hi, so I've got a workaround. @ #26 means we can pass an option to attach the component to the DOM when mount is called.

When a component has been mounted to the DOM, hasStyle will check the node without having to attach it to the DOM. 

The change to hasStyle hasn't been committed yet, but I'll update you when it's released. Should be by Saturday at the latest.

**Comment by matrunchyk** [comment]
wow looks great! Thank you for all you guys do here!

**Comment by eddyerburgh** [comment]
Hi, this is fixed in 1.11.0.

You'll need to pass the option attachToDocument when you mount the component:

```js
mount(temp1, {attachToDocument: true})
```

This attaches the component to the document, so `hasStyle` can check for properties passed by multiple selectors 😄

**Comment by matrunchyk** [comment]
Beautiful, thank you! :)
